packages/migration_tool/src/PortlandLabs/Concrete5/MigrationTool/Publisher/Block/StandardPublisher.php

```diff
@@ -35,13 +38,30 @@ public function publish(Batch $batch, $bt, Page $page, $area, BlockValue $value)
             }
             // Now we import the OTHER records.
             if ($b) {
+                $app = Application::getFacadeApplication();
+                /** @var Connection $connection */
+                $connection = $app->make(Connection::class);
+                $sm = $connection->getSchemaManager();
                 foreach ($records as $record) {
                     if (strcasecmp($record->getTable(), $bt->getController()->getBlockTypeDatabaseTable()) != 0) {
+                        $columns = $sm->listTableColumns($record->getTable());
                         $aar = new BlockRecord($record->getTable());
                         $aar->bID = $b->getBlockID();
                         foreach ($record->getData() as $key => $value) {
                             $result = $inspector->inspect($value);
-                            $aar->{$key} = $result->getReplacedValue();
+                            $value = $result->getReplacedValue();
+                            foreach ($columns as $column) {
+                                if ($column->getName() === $key) {
+                                    if ($column->getType()->getName() === Type::INTEGER) {
+                                        if ($column->getNotnull()) {
+                                            $value = (int) $value;
+                                        } else {
+                                            $value = $value === null ? null : (int) $value;
+                                        }
+                                    }
+                                }
```

Comment:
```suggestion
                                if (
                                    $column->getName() !== $key ||
                                    $column->getType()->getName() === Type::INTEGER
                                ) {
                                    continue;
                                }
                                
                                $value =  ($column->getNotNull() || $column !== null) : (int) $value : null;
                                break;
```

Using an early return (early continue) is more readable to me here, also worth breaking when the proper column is matched